src/ui/map.js

```diff
@@ -2289,6 +2289,7 @@ class Map extends Camera {
         this._canvas.addEventListener('webglcontextrestored', this._contextRestored, false);
         this._canvas.setAttribute('tabindex', '0');
         this._canvas.setAttribute('aria-label', 'Map');
+        this._canvas.setAttribute('role', 'application');
```

Comment:
The `application` role has some unique implications for users of assistive technology that you may or may not be familiar with, or otherwise have considered:

https://www.w3.org/TR/wai-aria/#application:

> When there is a need to create an element with an interaction model that is not supported by any of the WAI-ARIA _widget_ roles, authors **_MAY_** give that element role `application`. And, when a user navigates into an element with role `application`, assistive technologies that intercept standard input events **_SHOULD_** switch to a mode that passes most or all standard input events through to the web application.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Application_Role:

> ATs intercept almost all keyboard input and consume it themselves, letting nothing through to the browser or other user agent. To be able to interact with a web page, a standard set of widgets is recognized that, when pressing a certain key (usually the <kbd>Enter</kbd> key) this mode is switched off. The screen reader mode, often called _forms mode_ or _focus mode_, lets all keyboard input go through to the browser again. <kbd>Escape</kbd> is the most common way of switching back to _browse_ mode.
>
> The application role is designed to provide a means for widgets that are not part of the standard set to be accessible for direct interaction in ATs that use both browse and focus modes for interacting with web content.

> The <kbd>tab</kbd>, <kbd>Space</kbd> and <kbd>Enter</kbd> keys, as well as <kbd>Escape</kbd> , must be handled by the application.

Additional informative article: https://www.marcozehe.de/if-you-use-the-wai-aria-role-application-please-do-so-wisely/.

For the Maps for HTML Community Group's `<mapml-viewer>` custom component (which uses Leaflet), we chose [`region`](https://www.w3.org/TR/wai-aria-1.1/#region) instead (https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/151), because it is generic enough, in terms of ARIA roles, and doesn't change the standard behavior in conjunction to screen readers.